packaging/amazon-linux-ami/ecs-init.spec

```diff
@@ -107,48 +178,54 @@ rm -rf $RPM_BUILD_ROOT
 # record whether or not our service was running when docker is upgraded
 ecs_status=$(/sbin/status ecs 2>/dev/null || :)
 if grep -qF "start/" <<< "${ecs_status}"; then
-	/sbin/stop ecs >/dev/null 2>&1 || :
-	if [ "$1" -ge 1 ]; then
-		# write semaphore if this package is still installed
-		touch %{running_semaphore} >/dev/null 2>&1 || :
-	fi
+        /sbin/stop ecs >/dev/null 2>&1 || :
+        if [ "$1" -ge 1 ]; then
+                # write semaphore if this package is still installed
+                touch %{running_semaphore} >/dev/null 2>&1 || :
+        fi
 fi
 
 %triggerpostun -- docker
 # ensures that ecs-init is restarted after docker or ecs-init is upgraded
 if [ "$1" -ge 1 ] && [ -e %{running_semaphore} ]; then
-	/sbin/start ecs >/dev/null 2>&1 || :
-	rm %{running_semaphore} >/dev/null 2>&1 ||:
+        /sbin/start ecs >/dev/null 2>&1 || :
+        rm %{running_semaphore} >/dev/null 2>&1 ||:
 fi
 
+%post
+# Symlink the bundled ECS Agent at loadable path.
+ln -sf ecs-agent-v%{bundled_agent_version}.tar %{cache_dir}/ecs-agent.tar
+# Indicate reload from cached image is needed.
+echo 2 > %{cache_dir}/state
```

Comment:
Instead of writing `2` in `%post`, should we just install this file with the content as `2`?

Reply:
I don't see why not! I'll update the packaging.